### PR TITLE
Explicitly cast padding to float

### LINKED ISSUE #750: DataQualityDict.populate breaks when invoked with pad=True
Using a `DataQualityDict` to read a veto definer and populate the dictionary breaks when run with  `pad=True`, which is the default for this method. 

Example below:
```
In [1]: from gwpy.segments import DataQualityFlag, DataQualityDict, SegmentList, Segment

In [2]: import numpy as np

In [3]: start = 1186775580.

In [4]: end = start + 100.

In [5]: science = DataQualityFlag.query('L1:DCS-ANALYSIS_READY_C02', start, end, url='https://segments.ligo.org')

In [6]: vdf= '/home/thomas.massinger/git/veto-definitions/cbc/O2/H1L1-CBC_VETO_DEFINER_CLEANED_C02_O2_1164556817-23176801.xml'

In [7]: vetoes = DataQualityDict.from_veto_definer_file(vdf, start=start, end=end, format='ligolw', ifo='L1')

In [8]: vetoes.populate(source='https://segments.ligo.org', segments=science.active)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-565ca0a55839> in <module>()
----> 1 vetoes.populate(source='https://segments.ligo.org', segments=science.active)

/usr/lib/python2.7/site-packages/gwpy/segments/flag.pyc in populate(self, source, segments, pad, on_error, **kwargs)
   1590             self[key].active = tmp[key].active
   1591             if pad:
-> 1592                 self[key] = self[key].pad(inplace=True)
   1593                 if segments is not None:
   1594                     self[key].known &= segments

/usr/lib/python2.7/site-packages/gwpy/segments/flag.pyc in pad(self, *args, **kwargs)
    809             raise TypeError("unexpected keyword argument %r"
    810                             % list(kwargs.keys())[0])
--> 811         new.known = [(s[0]+start, s[1]+end) for s in self.known]
    812         new.active = [(s[0]+start, s[1]+end) for s in self.active]
    813         return new

TypeError: in method 'LIGOTimeGPS___add__', argument 2 of type 'LIGOTimeGPS *'
```

### PR DESCRIPTION
**Pull request by tjma12**
Both `lal.LIGOTimeGPS.__add__()` and `glue.lal.LIGOTimeGPS.__add__()` are incompatible with `numpy.int32` datatypes, so padding values had to be cast to float when initializing a `DataQualityFlag`. This recovers the intended functionality of `DataQualityFlag.pad()` with `self.padding` as the intended padding window. It also fixes the bug with `DataQualityFlag.populate()` and `DataQualityDict.populate()` using padding windows from veto definers.

Fixes #750

### CODE REVIEW

**Comment by alurban** [comment]
@tjma12 It looks like the CI build got hung up on a unit test for open data:

```
    def test_fetch_open_data(self):
        try:
            segs = self.TEST_CLASS.fetch_open_data(
                'H1_DATA', 946339215, 946368015)
        except (URLError, SSLError) as exc:
>           pytest.skip(str(e))
E           NameError: global name 'e' is not defined
gwpy/tests/test_segments.py:693: NameError
```

I suspect there's a typo on line 693 of [this file](https://github.com/tjma12/gwpy/blob/master/gwpy/tests/test_segments.py#L693) (i.e. `str(e)` should be `str(exc)`), but more importantly, I don't think this failure has anything to do with your PR. I know that at least one other user couldn't access LOSC open data earlier today due to Tuesday maintenance shenanigans.

tl;dr: I think @duncanmmacleod probably has to kick the CI build to run it again, and we can fix the typo in a future PR when we get around to it.

**Comment by duncanmmacleod** [comment]
@tjma12, the previous `NameError` was just a bug in raising a real problem with the tests:

```python
    @padding.setter
    def padding(self, pad):
        if pad is None:
            self._padding = (float(0), float(0))
        else:
>           self._padding = (float(pad[0]), float(pad[1]))
E           TypeError: float() argument must be a string or a number
```

Something is calling `pad` with `(None, None)`, so your new `float` casting is failing. Can you update the code to pass the tests.

File: gwpy/segments/flag.py [hunk 1]
```diff
@@ -345,13 +345,13 @@ def padding(self):
     @padding.setter
     def padding(self, pad):
         if pad is None:
-            self._padding = (0, 0)
+            self._padding = (0., 0.)
         else:
-            self._padding = (pad[0], pad[1])
+            self._padding = tuple([0. if p is None else float(p) for p in pad])
```
Comment (duncanmmacleod) [inline gwpy/segments/flag.py]: @tjma12, this can be simplified to

```python
    def padding(self, pad):
        if pad is None:
            pad = (None, None)
        self._padding = tuple(float(p or 0.) for p in pad)
```

can you (test and) make that change?

**Comment by tjma12** [comment]
@duncanmmacleod The Travis build that failed seems unrelated to my PR, to confirm this I ran the relevant unit test (https://travis-ci.org/gwpy/gwpy/jobs/377384953#L2505) on the command line and it passed. Can someone kickstart Travis to try again?

```
python setup.py test --addopts "gwpy/tests/test_table.py::TestGravitySpyTable::test_search"

...

================================================= test session starts ==================================================
platform linux2 -- Python 2.7.5, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/thomas.massinger/opt/gwpy-test-flags/bin/python
cachedir: .cache
rootdir: /home/thomas.massinger/git/gwpy, inifile: setup.cfg
collected 1 item 

gwpy/tests/test_table.py::TestGravitySpyTable::test_search PASSED

============================================== 1 passed in 10.58 seconds ===============================================
```

**Comment by duncanmmacleod** [comment]
Thanks @tjma12!
